### DISCUSSION TITLE
Calculating the Household in a 3-phase-setup

### DISCUSSION THREAD

**Discussion by mafischerhei** [nathanmarlor/foxess_modbus#462] [General]
As we know now, that the SINGLE phases of Grid Consumption R S T are not allowed to go negative in this integration, we can't use them to calculate the household on 3-phase setup.
we have to use Grid CT R S T, Inverter Power R S T and EPS Power R S T instead to calculate the Household .

I have a chint connected. Only one inverter (H3)

At the moment, this template sensor(Helpers->Create Template Sensor)  seems to work for my setup:

`
{% set gridct_r = states('sensor.foxess_grid_ct_r') | float(default=0) %}

{% set gridct_s = states('sensor.foxess_grid_ct_s') | float(default=0) %}
{% set gridct_t = states('sensor.foxess_grid_ct_t') | float(default=0) %}
{% set gridct = (gridct_r + gridct_s + gridct_t) %}
{% set gridimport = gridct | abs %}
{% set rpower = states('sensor.foxess_inv_power_r') | float(default=0) %}
{% set spower = states('sensor.foxess_inv_power_s') | float(default=0) %}
{% set tpower = states('sensor.foxess_inv_power_t') | float(default=0) %}
{% set power = rpower + spower + tpower %}
{% set eps_rpower = states('sensor.foxess_eps_power_r') | float (default=0) %}
{% set eps_spower = states('sensor.foxess_eps_power_s') | float (default=0) %}
{% set eps_tpower = states('sensor.foxess_eps_power_t') | float (default=0) %}
{% set eps_power = eps_rpower + eps_spower + eps_tpower %}
{% set result = gridimport + power + eps_power | round (3) %}
{{ result }}
`

**Comment by benbeton** [comment]
ty @mafischerhei  I've added that and will monitore / compare it tomorrow

**Reply by mafischerhei** [reply]
Note: In my calculation, I do not consider "Battery Charge Power" as Houseload. Has to be added in the calculation if someone has the point of view of being it a part of houseload.

**Reply by mafischerhei** [reply]
I made some changes to avoid subtracting when exporting and only use Inverter power Output (only positive values):


{% set gridct_r = states('sensor.foxess_grid_ct_r') | float(default=0) %}
{% set gridct_s = states('sensor.foxess_grid_ct_s') | float(default=0) %}
{% set gridct_t = states('sensor.foxess_grid_ct_t') | float(default=0) %}
{% set gridct = (gridct_r + gridct_s + gridct_t) %}
{% set gridimport = 0 if gridct > 0 else gridct | abs  %}
{% set rpower = states('sensor.foxess_inv_power_r') | float(default=0) %}
{% set spower = states('sensor.foxess_inv_power_s') | float(default=0) %}
{% set tpower = states('sensor.foxess_inv_power_t') | float(default=0) %}
{% set power = rpower + spower + tpower %}
{% set power = 0 if power < 0 else power | round (3) %}
{% set eps_rpower = states('sensor.foxess_eps_power_r') | float (default=0) %}
{% set eps_spower = states('sensor.foxess_eps_power_s') | float (default=0) %}
{% set eps_tpower = states('sensor.foxess_eps_power_t') | float (default=0) %}
{% set eps_power = eps_rpower + eps_spower + eps_tpower %}
{% set result = gridimport + power + eps_power %}
{% set result = 0 if result < 0 else result | round (3) %}
{{ result }}

**Reply by benbeton** [reply]
looking good for now, not 100% same as "sensor.foxess_load_power" from foxess cloud integration but I guess I can now switch to the local version :D 

cloud:
<img width="582" alt="image" src="https://github.com/nathanmarlor/foxess_modbus/assets/1405209/21181767-03df-4cc9-9a45-48ee2cd2f18b">

Modbus:
<img width="568" alt="image" src="https://github.com/nathanmarlor/foxess_modbus/assets/1405209/1ba1ee0c-67c6-4f11-95e9-f423ccbcf133">

**Reply by mafischerhei** [reply]
Cool. Nice to hear. Thanks for trying out

**Comment by mafischerhei** [comment]
Now that its getting sunnier and I#m exporting, I realized that I forgot to take into account to subtract that for the household.
I changed the code to take it into account. Would be nice to get feedback if it works for you.

{% set gridct_r = states('sensor.foxess_grid_ct_r') | float(default=0) %}
{% set gridct_s = states('sensor.foxess_grid_ct_s') | float(default=0) %}
{% set gridct_t = states('sensor.foxess_grid_ct_t') | float(default=0) %}
{% set gridct = (gridct_r + gridct_s + gridct_t) %}
{% set gridimport = 0 if gridct > 0 else gridct | abs  %}
{% set gridexport = 0 if gridct < 0 else gridct | abs  %}
{% set rpower = states('sensor.foxess_inv_power_r') | float(default=0) %}
{% set spower = states('sensor.foxess_inv_power_s') | float(default=0) %}
{% set tpower = states('sensor.foxess_inv_power_t') | float(default=0) %}
{% set power = rpower + spower + tpower %}
{% set power = 0 if power < 0 else power | round (3) %}
{% set eps_rpower = states('sensor.foxess_eps_power_r') | float (default=0) %}
{% set eps_spower = states('sensor.foxess_eps_power_s') | float (default=0) %}
{% set eps_tpower = states('sensor.foxess_eps_power_t') | float (default=0) %}
{% set eps_power = eps_rpower + eps_spower + eps_tpower %}
{% set result = gridimport + power + eps_power - gridexport %}
{% set result = 0 if result < 0 else result | round (3) %}
{{ result }}

**Reply by phil239** [reply]
Works fine for me. Thanks. By the way, has there ever been a solution for the broken load power values? The values I see are definitely wrong. Restarting the Chint smart meter also didn‘t solve the issue. Has anyone tried to reset the Chint via the menus (pw “701”). I fear that this only deletes the total values

![IMG_3252](https://github.com/nathanmarlor/foxess_modbus/assets/50781999/34998e1b-161b-44bc-ade0-f44daa4a1c07)

**Reply by horstvogel** [reply]
> Now that its getting sunnier and I#m exporting, I realized that I forgot to take into account to subtract that for the household. I changed the code to take it into account. Would be nice to get feedback if it works for you.
> 
> {% set gridct_r = states('sensor.foxess_grid_ct_r') | float(default=0) %} {% set gridct_s = states('sensor.foxess_grid_ct_s') | float(default=0) %} {% set gridct_t = states('sensor.foxess_grid_ct_t') | float(default=0) %} {% set gridct = (gridct_r + gridct_s + gridct_t) %} {% set gridimport = 0 if gridct > 0 else gridct | abs %} {% set gridexport = 0 if gridct < 0 else gridct | abs %} {% set rpower = states('sensor.foxess_inv_power_r') | float(default=0) %} {% set spower = states('sensor.foxess_inv_power_s') | float(default=0) %} {% set tpower = states('sensor.foxess_inv_power_t') | float(default=0) %} {% set power = rpower + spower + tpower %} {% set power = 0 if power < 0 else power | round (3) %} {% set eps_rpower = states('sensor.foxess_eps_power_r') | float (default=0) %} {% set eps_spower = states('sensor.foxess_eps_power_s') | float (default=0) %} {% set eps_tpower = states('sensor.foxess_eps_power_t') | float (default=0) %} {% set eps_power = eps_rpower + eps_spower + eps_tpower %} {% set result = gridimport + power + eps_power - gridexport %} {% set result = 0 if result < 0 else result | round (3) %} {{ result }}

Is this example without a Batterie? or what is eps Power?

**Comment by horstvogel** [comment]
Hello,

the Power Flox Card Plus calculates this directly, probably the difference to the template is that the DC power of the PV system is listed here for calculation. In the picture above the result from the template and below directly from the Power Flow card. 



therefore probably the difference??
thanks

H3 Batterie Setup
![image](https://github.com/nathanmarlor/foxess_modbus/assets/25108737/69ca54fc-ef0b-40a0-bb3c-22b1ce635b70)
